src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,115 @@
-import { PeopleFilters } from './PeopleFilters';
+import { FC, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { Loader } from './Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
 import { PeopleTable } from './PeopleTable';
+import { PeopleFilters } from './PeopleFilters';
+
+export const PeoplePage: FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoadError, setIsLoadError] = useState(false);
+  const [search] = useSearchParams();
+  const searchSex = search.get('sex');
+  const query = search.get('query') || '';
+  const choiceCenturies = search.getAll('centuries');
+  const sort = search.get('sort');
+  const order = search.get('order');
+
+  const loadPeople = async () => {
+    setIsLoading(true);
+
+    try {
+      const peopleFromServer = await getPeople();
+
+      setPeople(peopleFromServer);
+    } catch {
+      setIsLoadError(true);
+    }
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const visiblePeople = people.filter(({
+    name, sex, born,
+  }) => {
+    const filterSex = searchSex === sex || searchSex === null;
+
+    const lowerName = name.toLowerCase();
+    const lowerQuery = query.toLowerCase().trim();
+    const filterName = lowerName.includes(lowerQuery);
+
+    const personCenturies = Math.round(born / 100).toString();
+    const filterCenturies = choiceCenturies.length === 0
+      || choiceCenturies.includes(personCenturies);
+
+    return filterSex && filterName && filterCenturies;
+  });
+
+  if (sort) {
+    visiblePeople.sort((person1, person2) => {
+      switch (sort) {
+        case 'born':
+          return person1[sort] - person2[sort];
+        case 'died':
+          return person1[sort] - person2[sort];
+        case 'name':
+          return person1[sort].localeCompare(person2[sort]);
+        case 'sex':
+          return person1[sort].localeCompare(person2[sort]);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
```

Comment:
move it to the `visiblePeople`

Reply:
ya tebya ne ponimat))) How I should move if or sort to variable, I can't do it in the filter and after because I should do if

Reply:
You can move the sort variable into useMemo deps array and sort people only when it has changed. You just need to return a final value inside your useMemo callback to do it, so you can apply both filter and sort there.